### DISCUSSION TITLE
潜在表現の出力

### DISCUSSION THREAD

**Discussion by RRR-troisR** [espnet/espnet#4030] [Q&A]
VITSの音声合成時に textencoder の出力のうち、潜在表現の確率分布の標準偏差に相当する部分（logs？）も見れるようにしたいのですがどのようにコードを書き換えればいいでしょうか？ 
今は jupyterでText2Speechをimportして使って音声合成を行っています。

**Comment by kan-bayashi** [comment]
返信が遅くなりました。
少々面倒ですが、
https://github.com/espnet/espnet/blob/38939c2fc65c366a907c833a8bc7ecd126344c14/espnet2/gan_tts/vits/generator.py#L553
の返り値に
https://github.com/espnet/espnet/blob/38939c2fc65c366a907c833a8bc7ecd126344c14/espnet2/gan_tts/vits/generator.py#L455
このあたりの欲しい物を追加したのち、
https://github.com/espnet/espnet/blob/38939c2fc65c366a907c833a8bc7ecd126344c14/espnet2/gan_tts/vits/vits.py#L599
ここの返り値が増えるので
https://github.com/espnet/espnet/blob/38939c2fc65c366a907c833a8bc7ecd126344c14/espnet2/gan_tts/vits/vits.py#L611
ここのdictに入れて返すと
https://github.com/espnet/espnet/blob/38939c2fc65c366a907c833a8bc7ecd126344c14/espnet2/bin/tts_inference.py#L219
ここで帰ってきます。
